File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res=[[1]]
        for i in range(numRows-1):
            curr=res[-1]
            temp=[curr[0]]
            for j in range(0,len(curr)-1):
                temp.append(curr[j]+curr[j+1])
            temp.append(1)
            res.append(temp)
        return res

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res=new ArrayList<>();
        ArrayList<Integer> temp=new ArrayList<>();
        temp.add(1);
        res.add(temp);
        for(int i=0;i<numRows-1;i++)
        {
            ArrayList<Integer> curr=new ArrayList<>();
            curr.add(1);
            for(int j=0;j<temp.size()-1;j++)
            {
                curr.add(temp.get(j)+temp.get(j+1));

            }
            curr.add(1);
            res.add(curr);
            temp=curr;
        }
        return res;
    }
}